contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -344,10 +356,13 @@ pub fn execute_burn(
         return Err(ContractError::Unauthorized {});
     }
 
-    // make sure is expired
-    let expiration = STAGE_EXPIRATION.load(deps.storage, stage)?;
-    if !expiration.is_expired(&env.block) {
-        return Err(ContractError::StageNotExpired { stage, expiration });
+    // make sure is expired if the stage is not paused
+    let is_paused = STAGE_PAUSED.load(deps.storage, stage)?;
```

Comment:
Instead of branching here, you can return `ContractError::Paused` to simplify the code